ui/src/shared/components/inlineLabels/InlineLabelsEditor.tsx

```diff
@@ -159,52 +153,57 @@ class InlineLabelsEditor extends Component<Props, State> {
   }
 
   private handleDismissPopover = () => {
-    const {labels: filteredLabels} = this.props
-
-    this.setState({isPopoverVisible: false, filteredLabels})
+    this.setState({isPopoverVisible: false})
   }
 
-  private handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
+  private handleInputChange = async (
+    e: ChangeEvent<HTMLInputElement>
+  ): Promise<void> => {
+    const {availableLabels} = this
     let selectedItemID = this.state.selectedItemID
-    const {labels, selectedLabels} = this.props
     const searchTerm = e.target.value
 
-    const availableLabels = _.differenceBy(labels, selectedLabels, l => l.name)
+    const selectedItemIDAvailable = this.availableLabels.find(
+      al => al.name === selectedItemID
+    )
 
-    const filteredLabels = availableLabels.filter(label => {
+    const matchingLabels = availableLabels.filter(label => {
       return label.name.includes(searchTerm)
     })
 
-    const selectedItemIDAvailable = filteredLabels.find(
-      al => al.name === selectedItemID
-    )
-
-    if (!selectedItemIDAvailable && filteredLabels.length) {
-      selectedItemID = filteredLabels[0].name
+    if (!selectedItemIDAvailable && matchingLabels.length) {
+      selectedItemID = matchingLabels[0].name
     }
 
     if (searchTerm.length === 0) {
-      return this.setState({
-        filteredLabels: this.props.labels,
-        selectedItemID: null,
-        searchTerm: '',
-      })
+      selectedItemID = null
     }
 
-    this.setState({searchTerm, filteredLabels, selectedItemID})
+    this.setState({searchTerm, selectedItemID})
+  }
+
+  private get filteredLabels(): ILabel[] {
+    const {searchTerm} = this.state
+    const {labels, selectedLabels} = this.props
+
+    const availableLabels = _.differenceBy(labels, selectedLabels, l => l.name)
+
+    return availableLabels.filter(label => {
+      return label.name.includes(searchTerm)
+    })
   }
 
   private get availableLabels(): ILabel[] {
-    const {selectedLabels} = this.props
-    const {filteredLabels} = this.state
+    const {selectedLabels, labels} = this.props
 
-    return _.differenceBy(filteredLabels, selectedLabels, label => label.name)
+    return _.differenceBy(labels, selectedLabels, label => label.name)
   }
 
   private handleCreateLabel = async (label: ILabel) => {
     const {onCreateLabel, onAddLabel} = this.props
-    const newLabel = await onCreateLabel(label)
-    onAddLabel(newLabel)
+    await onCreateLabel(label)
```

Comment:
should these be in try catches, with notifications for failures?

Reply:
I'll create issues for adding notifications, but there should be some and there are `// Todo` comments referring to them in resource cards.